### ISSUE TITLE
Add support for external textures

### ISSUE THREAD

**Issue by bibendovsky** [bibendovsky/bstone#316]

**Comment by bibendovsky** [comment]
@NJW1979

> When adding HD texture support can you have the port use the same texture names for HD textures as the names of the extracted textures? This way I don't have to rename the HD textures for the port to pick them up.

Yes, just place them in appropriate directory (see *Naming conventions*).

> P.S. Would the port automatically stretch to the different aspect ratio, or do my textures have to be in that ratio to work?

You should work with 1:1.2 but export to 1:1 for maximum compatibility / performance.
Some hardware may have degraded perfomance if texture has non-power-of-two dimensions.

**Comment by NJW1979** [comment]
Thanks for the info.  I look forward to HD texture support so I can make a texture pack for both AOG and PS.  It's a much simpler job than the one I am working on for Doom.

**Comment by bibendovsky** [comment]
Here are Windows builds with external textures support.
Use `GAME OPTIONS - VIDEO - TEXTURING - EXTERNAL TEXTURES` option to toggle the textures.

[bstone_bin_win32_v1_2_10_beta_202101091420.zip](https://github.com/bibendovsky/bstone/files/5791184/bstone_bin_win32_v1_2_10_beta_202101091420.zip)
[bstone_bin_win64_v1_2_10_beta_202101091420.zip](https://github.com/bibendovsky/bstone/files/5791185/bstone_bin_win64_v1_2_10_beta_202101091420.zip)

**Comment by NJW1979** [comment]
Thanks for the update.  I already made a doorjamb texture and it works perfectly.
https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/ded1fd6b-6738-42b8-b0cf-d5fd40ca7500/debyi4z-130175f4-caa7-43a1-bd67-7c1776288e9d.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3sicGF0aCI6IlwvZlwvZGVkMWZkNmItNjczOC00MmI4LWIwY2YtZDVmZDQwY2E3NTAwXC9kZWJ5aTR6LTEzMDE3NWY0LWNhYTctNDNhMS1iZDY3LTdjMTc3NjI4OGU5ZC5wbmcifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6ZmlsZS5kb3dubG9hZCJdfQ.5Iereeahu4UCCzycaXHE_QascNSrsOGakyA3ehJLlRc

**Comment by bibendovsky** [comment]
Red circles on the door looks stretched. Is it intentional or it is just a work-in-progress image?

**Comment by NJW1979** [comment]
I think the issue might be the port slightly stretching the textures.  ECWolf does the same in Wolfenstein 3D. maybe if you add an aspect option for the textures so it can widen them until they are a true square.  That would solve the stretching issue 

This is how the door should look.
https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/ded1fd6b-6738-42b8-b0cf-d5fd40ca7500/dec0vg1-e9225bf9-0ff5-484b-8506-83beb93e0245.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3sicGF0aCI6IlwvZlwvZGVkMWZkNmItNjczOC00MmI4LWIwY2YtZDVmZDQwY2E3NTAwXC9kZWMwdmcxLWU5MjI1YmY5LTBmZjUtNDg0Yi04NTA2LTgzYmViOTNlMDI0NS5wbmcifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6ZmlsZS5kb3dubG9hZCJdfQ.oL27Qw0YApYKxZuR2P-Kt7fshjwhVe1wDNJCfQEvs1Q

**Comment by bibendovsky** [comment]
> I think the issue might be the port slightly stretching the textures. ECWolf does the same in Wolfenstein 3D. maybe if you add an aspect option for the textures so it can widen them until they are a true square. That would solve the stretching issue

Such option is useless - the geometry of wall or sprite in the port is not square (flooring and ceiling are not affected).

Solutions for non-square geometry:
1. Design in 1:1.2 but export in 1:1. Requires the artist to shrink the image vertically by 20% on export.
2. Design in 1:1.2 and export in 1:1.2. Ideal for artist but may have negative impact on performance.
3. Design in 1:1 and export in 1:1. Requires the artist to design content already shrinked vertically by 20%.

**Comment by NJW1979** [comment]
One thing I found out with that door texture is it only stretches when facing it dead on.  When I view the texture from an angle the red lights are perfect circles.
